### PR TITLE
Implement Get mentor by id endpoint (Public endpoint)

### LINKED ISSUE #15: Implement Get mentor by id endpoint (Public endpoint)
**Description:**
This issue involves implementing an API to get mentors by id. The endpoint should allow clients to make a GET request to `{{baseUrl}}/api/mentee/mentors/{{mentorId}}`.

```
{
    "mentorId": 5,
    "category": "Computer Science",
    "profile": {
        "contact_email": "mandy@gmail.com",
        "first_name": "Mandy",
        "last_name": "Fernando",
        "image_url": "https://www.linkedin.com/mandy",
        "linkedin_url": "https://www.linkedin.com/mandy"
    },
    "mentees": []
}
```

**Tasks:**
1. Create a controller for the `/controllers/mentor` endpoint in the backend (create the route mentor if not created).
2. Parse and validate the request body to ensure it matches the expected format.
3. Implement appropriate error handling and response status codes for different scenarios (e.g., validation errors, database errors).
5. Write unit tests to validate the functionality and correctness of the endpoint.

API documentation: https://documenter.getpostman.com/view/27421496/2s93m1a4ac#8744a3ee-970f-489a-853d-8b23fdee8de3

ER diagram: https://drive.google.com/file/d/11KMgdNu2mSAm0Ner8UsSPQpZJS8QNqYc/view 

**Acceptance Criteria:**
- The profile API endpoint `/controllers/mentor` is implemented and accessible via a GET request.
- The request body is properly parsed and validated for the required format.
- Appropriate error handling is implemented, providing meaningful error messages and correct response status codes.
- Unit tests are written to validate the functionality and correctness of the endpoint.

**Additional Information:**
No

**Related Dependencies or References:**
No

### PR DESCRIPTION
**Pull request by Madhawa97**
## Purpose

The purpose of this PR is to fix #15

## Goals

This issue involves implementing an API to get mentors by id. The endpoint should allow clients to make a GET request to `{{baseUrl}}/api/mentors/{{mentorId}}`.
```
{
    "mentorId": 5,
    "category": "Computer Science",
    "profile": {
        "contact_email": "mandy@gmail.com",
        "first_name": "Mandy",
        "last_name": "Fernando",
        "image_url": "https://www.linkedin.com/mandy",
        "linkedin_url": "https://www.linkedin.com/mandy"
    },
    "mentees": []
}
```

## Approach

- Create a controller for the /controllers/mentor endpoint in the backend (create the route mentor if not created).
- Implement appropriate error handling and response status codes for different scenarios (e.g., validation errors, database errors).
- Write unit tests to validate the functionality and correctness of the endpoint.

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Test environment
 
macOS 13 latest, local Postgres 15 env.

### CODE REVIEW

**Comment by Madhawa97** [comment]
Please verify if the request URL is `{{baseUrl}}/api/mentors/{{mentorId}}`. Currently, I've used that. It has given as `{{baseUrl}}/api/mentees/mentors/{{mentorId}}` in the API documentation.

**Comment by anjula-sack** [comment]
> Please verify if the request URL is `{{baseUrl}}/api/mentors/{{mentorId}}`. Currently, I've used that. It has given as `{{baseUrl}}/api/mentees/mentors/{{mentorId}}` in the API documentation. cc: @anjula-sack @janithlahirukariyawasam

Sorry, my bad.  I updated the doc. It's `{{baseUrl}}/api/mentors/{{mentorId}}`

**Comment by Madhawa97** [comment]
Thank you @anjula-sack.

File: src/controllers/mentor.controller.ts [hunk 1]
```diff
@@ -26,3 +26,38 @@ export const mentorApplicationHandler = async (
...
+        mentorId: mentor.uuid,
+        category: mentor.category.category,
+        profile: {
+          contact_email: mentor.profile.contact_email,
+          first_name: mentor.profile.first_name,
+          last_name: mentor.profile.last_name,
+          image_url: mentor.profile.image_url,
+          linkedin_url: mentor.profile.linkedin_url
+        },
+        mentees: mentor.mentees
```
Comment (anjula-sack) [inline src/controllers/mentor.controller.ts:51]: We don't need this.

```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/controllers/mentor.controller.ts (hunk 1 above)
Comment (Madhawa97) [reply src/controllers/mentor.controller.ts:51]: Okay

File: src/routes/mentor/mentor.route.test.ts [hunk 2]
```diff
@@ -60,6 +63,23 @@ describe('Mentor application', () => {
...
+  describe('Get mentor details route', () => {
+    it('should return a 200 with the mentor details', async () => {
+      const response = await agent
+        .get(`/api/mentors/${savedMentor.uuid}`)
+        .expect(200)
+
+      expect(response.body).toHaveProperty('mentorId')
+      expect(response.body).toHaveProperty('category')
+      expect(response.body).toHaveProperty('profile')
+    })
```
Comment (anjula-sack) [inline src/routes/mentor/mentor.route.test.ts:77]: Write a test to call the API unauthenticated as well @Madhawa97

File: src/controllers/mentor.controller.ts [hunk 3]
```diff
@@ -26,3 +26,37 @@ export const mentorApplicationHandler = async (
...
+        category: mentor.category.category,
+        profile: {
+          contact_email: mentor.profile.contact_email,
+          first_name: mentor.profile.first_name,
+          last_name: mentor.profile.last_name,
+          image_url: mentor.profile.image_url,
+          linkedin_url: mentor.profile.linkedin_url
+        }
+      }
+      res.status(statusCode).json({ ...mentorDetails })
```
Comment (anjula-sack) [inline src/controllers/mentor.controller.ts:52]:
```suggestion
           res.status(statusCode).json({ ...mentorDetails })
      }
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/routes/mentor/mentor.route.test.ts (hunk 2 above)
Comment (Madhawa97) [reply src/routes/mentor/mentor.route.test.ts:77]: Sure!

File: src/controllers/mentor.controller.ts (hunk 3 above)
Comment (Madhawa97) [reply src/controllers/mentor.controller.ts:52]: This is already formatted. Isn't it?

**Comment by Madhawa97** [comment]
Fixed merge conflicts with #9.

File: src/controllers/mentor.controller.ts (hunk 3 above)
Comment (anjula-sack) [reply src/controllers/mentor.controller.ts:74]: Let's update this like this @Madhawa97 

```
    if (!mentor) {
      return res.status(statusCode).json({ error: message });
    }

    const mentorDetails: MentorDetailsResponse = {
      mentorId: mentor.uuid,
      category: mentor.category.category,
      profile: {
        contact_email: mentor.profile.contact_email,
        first_name: mentor.profile.first_name,
        last_name: mentor.profile.last_name,
        image_url: mentor.profile.image_url,
        linkedin_url: mentor.profile.linkedin_url,
      },
    };

    return res.status(statusCode).json({ ...mentorDetails });
   ```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Review by anjula-sack — APPROVED** [review]

**Pull request merged**